src/evict/evict_lru.c

```diff
@@ -909,20 +940,20 @@ __evict_lru_walk(WT_SESSION_IMPL *session)
 		}
 	}
 
-	queue->evict_current = queue->evict_queue;
-	__wt_spin_unlock(session, &queue->evict_lock);
-
-	/*
-	 * Now we can set the next queue.
-	 */
-	__wt_spin_lock(session, &cache->evict_queue_lock);
-	if (cache->evict_current_queue->evict_current == NULL)
+	if (__evict_queue_empty(queue)) {
+		/*
+		 * This score varies between 0 (if the queue hasn't been empty
+		 * for a long time) and 100 (if the queue has been empty the
+		 * last 10 times we filled up.
+		 */
+		cache->evict_empty_score = WT_MIN(100,
+		    cache->evict_empty_score + WT_EVICT_EMPTY_SCORE_BUMP);
```

Comment:
A few things here:
1.  The comment describing the meaning of the score is buried here.  Maybe move it to `cache.h`?
2.  The conditional change is confusing or possibly wrong.  The original conditional checked if the previous `cache->evict_current_queue` was empty before replacing it with the newly filled queue.  The new conditional checks if the newly filled `queue` is empty.
3.  I think the entire, fixed, conditional and setting of the stats needs to move to where `cache->evict_current_queue` is set.  We're interested in knowing if the last batch of pages were fully processed or not.
4.  This code now has two locations where `cache->evict_current_queue` is set.  Previously there was only one place, here at the bottom of this function.  Now it is earlier in this fucntion and also in `evict_get_ref`.  That seems like it may end up error-prone.